docs/11.0/guides/upgrade-and-migration/migrating-from-10-to-11.md

```diff
@@ -0,0 +1,57 @@
+---
+title: Migrating from 10.0 to 11.0
+metaTitle: Migrating from 10.0 to 11.0 - Guide - Handsontable Documentation
+permalink: /next/migration-from-10.0-to-11.0
+canonicalUrl: /migration-from-10.0-to-11.0
+pageClass: migration-guide
+---
+
+# Migrating from 10.0 to 11.0
+
+[[toc]]
+
+To upgrade your Handsontable version from 10.x.x to 11.x.x, follow this guide.
+
+## Step 1: React, Angular, Vue – register your modules
+
+Starting with Handsontable 11.0.0, the [React wrapper](@/guides/integrate-with-react/react-installation.md), the [Angular wrapper](@/guides/integrate-with-angular/angular-installation.md), and the [Vue wrapper](@/guides/integrate-with-vue/vue-installation.md) support [modularization](@/guides/building-and-testing/modules.md).
+
+If you don't use any of the wrappers, you don't need to change anything.
+
+#### Using all modules
+
+To continue using all Handsontable modules with your wrapper, register all modules with the new `registerAllModules()` method.
+
+In the entry point file of your application, add the following code:
+```js
+// import the registerAllModules() method
+import { registerAllModules } from 'handsontable/registry';
+
+// register all Handsontable modules
+registerAllModules();
+```
+
+#### Using individual modules
+
+To start using individual Handsontable modules with your wrapper, see the following guides:
+- [Using modules with React](@/guides/integrate-with-react/react-modules.md)
+- [Using modules with Angular](@/guides/integrate-with-angular/angular-modules.md)
+- [Using modules with Vue](@/guides/integrate-with-vue/vue-modules.md)
+
+## Step 2: Adapt to the TypeScript definition file changes
+
+Before, all of Handsontable's TypeScript definitions were kept in one huge file, placed in the root directory: `./handsontable.d.ts`.
+
+Now, each module has its own TypeScript definitions file. They're all kept in a new `types` directory: [`./handsontable/types`](https://github.com/handsontable/handsontable/tree/master/handsontable/types).
+
+For more details, see [this pull request](https://github.com/handsontable/handsontable/pull/8875).
+
+## Step 3: Adapt to the `populateFromArray()` method's changes
```

Comment:
This section has no "migration" action.

`beforeChange` and `afterChange` hooks are not mentioned in the guide. Should they @wszymanski? Do we have any migration proposition for users of previous behavior?

Reply:
The method will give the same effect, but underneath implementation changed. It is impossible to give exact solution and for me it's hard to show even an example. I'm afraid that we can just describe what changed.

So far, there has been triggered `n` `beforeChange` and `afterChange` hooks, where `n` is number of rows/columns. From now, there will be triggered just one `beforeChange` and `afterChange`  hook.

Reply:
@wojciechczerniak @wszymanski - how about this? ee20b1a

Reply:
> So far, there has been triggered n beforeChange and afterChange hooks, where n is number of rows/columns. From now, there will be triggered just one beforeChange and afterChange hook. And maybe extra needed information is that source of the change for that action won't be spliceCol/spliceRow anymore, but generally populateFromArray.

Ok, then this description should be added. Maybe with a note that migration depends on implementation and should be considered individually. 

I have an idea for an example for `source` breaking change:

Before
```ts
new Handsontable(element, {
  afterChange: (changes, source) => {
    if (source === 'spliceRow' || source === 'spliceCol') {
       // you code ...
    }
  }
})
```

after:
```ts
new Handsontable(element, {
  afterChange: (changes, source) => {
    if (source === 'populateFromArray') {
       // you code ...
    }
  }
})
```

Reply:
And an example for `n-changes` triggered for `afterChange` and `beforeChange`:

Before, there were always one item in `changes` 
```ts
new Handsontable(element, {
  afterChange: (changes) => {
    handleChange(changes[0])
  }
})
```

After, there are multiple changes in `changes` and should be handled in a loop:
```ts
new Handsontable(element, {
  afterChange: (changes) => {
    changes.forEach(change =>  handleChange(change))
  }
})
```

WDYT @wszymanski? A broad idea, I imagine you could polish it a little

Reply:
> I have an idea for an example for `source` breaking change:

Ok, it may be helpful.

> And an example for `n-changes` triggered for `afterChange` and `beforeChange`:

I would not add it. Generally, `changes` argument is always an array. That's why it's hard to show that something changed.

Reply:
> @wojciechczerniak @wszymanski - how about this? [ee20b1a](https://github.com/handsontable/handsontable/commit/ee20b1abd70413d506e8621f85ec86b69cd7918c)

https://github.com/handsontable/handsontable/pull/8931#pullrequestreview-807257047

Reply:
@wojciechczerniak @wszymanski - added: [11](https://github.com/handsontable/handsontable/blob/release/11.0.0-docs/docs/11.0/guides/upgrade-and-migration/migrating-from-10.0-to-11.0.md#step-3-adapt-to-the-populatefromarray-methods-changes) and [next](https://github.com/handsontable/handsontable/blob/release/11.0.0-docs/docs/next/guides/upgrade-and-migration/migrating-from-10.0-to-11.0.md#step-3-adapt-to-the-populatefromarray-methods-changes)

Reply:
@kirszenbaum @wojciechczerniak I've [edited above comment.](https://github.com/handsontable/handsontable/pull/8931#discussion_r750231245). Wojciech's suggestion looks good.

Reply:
@wszymanski Looks good 👍 I would keep `source === 'spliceRow' || source === 'spliceColumn'`. Because `source === 'populateFromArray'` covers both. To differentiate between spliceRow and spliceColumn we would need additional logic

Reply:
@wszymanski - no prob, thanks! Updated: f2bfd4b caa833a

Reply:
@wszymanski @wojciechczerniak - brought back `source === 'spliceRow' || source === 'spliceColumn'`, as suggested: 45f2c20